File: 01/a/main.py
import sys

x, y = [], []

with open(sys.argv[1]) as f:
    for line in f:
        x_str, y_str = line.strip().split()
        x.append(int(x_str))
        y.append(int(y_str))

x.sort()
y.sort()

print(sum(abs(xx - yy) for xx, yy in zip(x, y)))

File: 01/a/main.cpp
#include <algorithm>
#include <cmath>
#include <fstream>
#include <functional>
#include <numeric>
#include <print>
#include <sstream>
#include <string>
#include <vector>

int main(const int, const char** argv) {
  std::vector<int> x, y;
  std::ifstream file{argv[1]};
  std::string line;
  int i, j;

  while (std::getline(file, line)) {
    std::stringstream ss;
    ss << line;
    ss >> i >> j;
    x.push_back(i);
    y.push_back(j);
  }

  std::sort(std::begin(x), std::end(x));
  std::sort(std::begin(y), std::end(y));

  auto answer{std::transform_reduce(
    std::cbegin(x),
    std::cend(x),
    std::cbegin(y),
    0,
    std::plus<>(),
    [](const auto first, const auto second) { return std::abs(first - second); }
  )};

  std::println("{}", answer);
}
